### ISSUE TITLE
I am getting error when I use multipleSelect . "Cannot find type definition file for 'trusted-types'. "

### ISSUE THREAD

**Issue by roopeshkurian** [ghiscoding/Angular-Slickgrid#1313]
### Describe the bug

I attempted to upgrade Angular-Slickgrid to the most recent version. Then I tried to run/build the project. But its giving me the following errors. 

Error: node_modules/multiple-select-vanilla/dist/types/interfaces/interfaces.d.ts:1:23 - error TS2688: Cannot find type definition file for 'trusted-types'.    

1 /// <reference types="trusted-types" />
                        ~~~~~~~~~~~~~


Error: node_modules/multiple-select-vanilla/dist/types/interfaces/interfaces.d.ts:47:44 - error TS2304: Cannot find name 'TrustedHTML'.

                                              ~~~~~~~~~~~
                                              ~~~~~~~~~~~


Error: node_modules/multiple-select-vanilla/dist/types/interfaces/multipleSelectOption.interface.d.ts:1:23 - error TS2688: Cannot find type definition file for 'trusted-types'.

1 /// <reference types="trusted-types" />
                        ~~~~~~~~~~~~~

Error: node_modules/multiple-select-vanilla/dist/types/interfaces/multipleSelectOption.interface.d.ts:174:44 - error TS2304: Cannot find name 'TrustedHTML'.    

174     sanitizer?: (html: string) => string | TrustedHTML;
                                               ~~~~~~~~~~~it is

### Reproduction

If you upgrade to the latest version, you may encounter this error.  If you did not get this issue, I can create a sample project for your reference.

### Expectation

There should not be any error from the node_modules folder. 

### Environment Info

```shell
Angular 17.0.2
Angular-Slickgrid 6.5.0
Typescript 5.2.2
Browsers: Mozilla, Chrome, Edge
System Windows 11
```


### Validations

- [X] Follow our [Code of Conduct](https://github.com/ghiscoding/Angular-Slickgrid/blob/master/.github/CODE_OF_CONDUCT.md)
- [X] Read the [HOWTO - Step by Step](https://github.com/ghiscoding/angular-slickgrid/wiki/HOWTO---Step-by-Step).
- [X] Check that there isn't [already an issue](https://github.com/ghiscoding/Angular-Slickgrid/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/ghiscoding/Angular-Slickgrid/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by roopeshkurian** [comment]
After I installed @types/dompurify, everything started working properly again.

**Comment by ghiscoding** [comment]
oh ok thanks for the feedback, I will make `@types/dompurify` a dependency of `multiple-select-vanilla` lib which I had a new release last week. I'll do that tomorrow, thanks for letting me know

**Comment by ghiscoding** [comment]
There was a request in TypeScript to make `trusted-types` part of the DOM API but it's been open for couple years without much traction in this TypeScript [issue](https://github.com/microsoft/TypeScript/issues/30024). Can you try `@types/trusted-types` instead? I think I will use that type instead of the DOMPurify one since that is what they are suggesting to use in that TypeScript issue.

**Comment by roopeshkurian** [comment]
I have removed DOMPurify and installed @types/trusted-types.  Its working well.  Thank you for your suggestion.

**Comment by ghiscoding** [comment]
Great thanks again for confirming, I will release a new version tomorrow to include it has a dependency so you won't need to install it yourself. :)

**Comment by roopeshkurian** [comment]
Thank you so much for your support.  It will be really helpful.

I have another request (Not important).
We are using momentjs . Can we think about implementing Luxon instead of Moment?

**Comment by ghiscoding** [comment]
> We are using momentjs . Can we think about implementing Luxon instead of Moment?

No I tried other lib like DayJS and date-fns to replace moment but they are all missing a little something, most of them don't have the `Z` for Zulu for UTC date. 2 years ago I decided to wait for [Temporal](https://github.com/tc39/proposal-temporal) to come out which will become part of EcmaScript and I tought it would have been out 2 years ago but it is really slow to finish that implementation but it does look like it's finally coming in the next few months since I think they finally closed the big roadblock they had, I'm watching this Temporal [roadmap](https://github.com/tc39/proposal-temporal/issues/2628), I'm hopefull it will be part of ES2024

And by the way, I'm using [moment-mini](https://www.npmjs.com/package/moment-mini) instead of `moment` because it's a lot smaller since it only includes the English locale and that is ok for my libs since I only use strict date format and I'm not using any of its locale functionalities. 

So in short, I'm not interested to move to other libs, I wanted to do that couple years ago but didn't find good libraries to migrate to since they are all missing something. However at this point `Temporal` is coming really soon and it will be part of the browser eventually, so I think that is the way to go, we just need to be patient and wait. However, I know that for the beginning we'll have to use a polyfill but I still prefer to wait for the official `Temporal` release or what they call Stage 4 (it's currently at Stage 3), hopefully soon.

**Comment by roopeshkurian** [comment]
I agree with you. Thank you for providing a thorough explanation.

**Comment by ghiscoding** [comment]
closed by PR #1314
